libs/MobileSync/src/com/salesforce/androidsdk/mobilesync/target/CollectionSyncUpTarget.kt

```diff
@@ -0,0 +1,139 @@
+     */
+    /**
+     * Construct CollectionSyncUpTarget
+     */
+    /**
+     * Construct CollectionSyncUpTarget
+     */
+    /**
+     * Construct CollectionSyncUpTarget with given id/modifiedDate/externalId fields
+     */
+    @JvmOverloads
+    constructor(
+        createFieldlist: List<String>? = null,
+        updateFieldlist: List<String>? = null,
+        idFieldName: String? = null,
+        modificationDateFieldName: String? = null,
+        externalIdFieldName: String? = null,
+        maxBatchSize: Int = MAX_RECORDS_SOBJECT_COLLECTION_API
+    ) : super(
+        createFieldlist,
+        updateFieldlist,
+        idFieldName,
+        modificationDateFieldName,
+        externalIdFieldName
+    ) {
+        this.maxBatchSize = min(
+            maxBatchSize,
+            MAX_RECORDS_SOBJECT_COLLECTION_API
+        ) // soject collection apis allows up to 200 records
+    }
+
+    /**
+     * Construct SyncUpTarget from json
+     *
+     * @param target
+     * @throws JSONException
+     */
+    constructor(target: JSONObject) : super(target) {
+        maxBatchSize = min(
+            target.optInt(
+                MAX_BATCH_SIZE,
+                MAX_RECORDS_SOBJECT_COLLECTION_API
+            ), MAX_RECORDS_SOBJECT_COLLECTION_API
+        ) // soject collection apis allows up to 200 records
+    }
+
+    @Throws(JSONException::class, IOException::class)
+    override fun sendRecordRequests(
+        syncManager: SyncManager,
+        recordRequests: List<RecordRequest>
+    ): Map<String, RecordResponse> {
+        return CompositeRequestHelper.sendAsCollectionRequests(syncManager, false, recordRequests)
+    }
+
+    @Throws(JSONException::class, IOException::class)
+    override fun areNewerThanServer(
+        syncManager: SyncManager,
+        records: List<JSONObject>
+    ): MutableMap<String, Boolean> {
+        val storeIdToNewerThanServer: MutableMap<String, Boolean> = HashMap()
+        val nonLocallyCreatedRecords: MutableList<JSONObject> = ArrayList()
+        for (record in records) {
+            if (isLocallyCreated(record) || !record.has(idFieldName)) {
+                val storeId = record.getString(SmartStore.SOUP_ENTRY_ID)
+                storeIdToNewerThanServer[storeId] = true
+            } else {
+                nonLocallyCreatedRecords.add(record)
+            }
+        }
+        val recordIdToRemoteModDate = fetchLastModifiedDates(syncManager, nonLocallyCreatedRecords)
+        for (record in nonLocallyCreatedRecords) {
+            val storeId = record.getString(SmartStore.SOUP_ENTRY_ID)
+            val localModDate = RecordModDate(
+                JSONObjectHelper.optString(record, modificationDateFieldName),
+                isLocallyDeleted(record)
+            )
+            val remoteModDate = recordIdToRemoteModDate[storeId]
+            storeIdToNewerThanServer[storeId] = isNewerThanServer(localModDate, remoteModDate!!)
```

Comment:
I also wasn't sure about a way to resolve this force unwrap, but took a look.  If the remote modified date is somehow missing, is there a default action rather than let this throw? For instance, does that record actually need an update to cleanse the missing date value, so that value in the map should be true?  Just thinking of ideas since the force unwrap is the same as throwing an exception on null.  Perhaps, if this is a real guard, could this logic throw a more detailed exception than the default unwrap provided by Kotlin?